components/button/src/button/button.js

```diff
@@ -122,14 +122,13 @@ Button.propTypes = {
     name: PropTypes.string,
     /**
      * Applies 'primary' button appearance.
-     * Mutually exclusive with `destructive` and `secondary` props
+     * Mutually exclusive with `destructive` prop
      */
     primary: sharedPropTypes.buttonVariantPropType,
     /**
      * Applies 'secondary' button appearance.
-     * Mutually exclusive with `primary` and `destructive` props
      */
-    secondary: sharedPropTypes.buttonVariantPropType,
+    secondary: PropTypes.bool,
```

Comment:
That's true. It doesn't make really make sense, conceptually. Ideally there would be some way to say "a `primary` button cannot have the `secondary` modifier applied to it", but I'm not sure that's possible without restructuring props?

Reply:
I see three ways to resolve this:

1. We introduce a breaking change and don't use boolean props but something like `priority="primary|secondary"` and `severity="normal (default)|destructive"
  - Pro: It'd make it very clear and we wouldn't have to use custom prop types to check if the user provided sane button configuration
  - Contra: This would be a breaking change
1. We implement another custom prop type
  - Pro: This would make the current implementation work
  - Contra: It'd make it more complex
  - Contra: It'd be hard to understand which configuration you can use and which is valid/invalid
1. We don't check whether the consumer provides sane props
  - Pro: We don't have to validate whether the button configuration is sane
  - Pro: It's very clear what prop type a prop has (they're all just booleans)
  - Contra: Consumers of the button component could provide a wrong configuration

I'd be in favor of simply using boolean prop types. The introduce the least amount of complexity, each prop by itself is quite simple (it's a boolean). At the same time, if the usage is wrong the developer should notice it during development. If they don't see it, then it should be the QA team who notices. If they don't see it either, ideally we'll get a bug ticket if the misuse is causing an issue. In any way, I'd rather trust the developers / the reporting process than introducing complexity to the library (in this case there the complexity is already there and we see now that it makes things harder because we have to talk about this)

Reply:
Option 3 is fine with me. I don't think removing the enforcement of exclusivity for the button type props will cause any real issues that aren't easily fixed (i.e. you need to review the button props if you've set `destructive` but it still appears `primary`).

Option 1 would've been my choice if this was a new component, but I don't think it's worth the breaking change.

Reply:
@Mohammer5, I've updated this PR to remove the `Button` props from `SharedPropTypes`, replacing them with simple booleans.